frontend/components/SignupForm/SignupForm.tsx

```diff
@@ -1,98 +1,90 @@
 import React from 'react'
 import Link from 'next/link'
+import { useRouter } from 'next/router'
+import { useForm, ErrorMessage } from 'react-hook-form'
 import { trackCreateAccount } from '../../events/users'
-import validateAuth, { IErrors } from '../../lib/validateAuth'
 import { useCreateUserMutation } from '../../generated/graphql'
-import useFormValidation from '../../hooks/useFormValidation'
 import FormError from '../FormError'
 import Button from '../../elements/Button'
 import { brandBlue } from '../../utils'
-
-interface IFormValues {
-  handle: string
-  email: string
-  password: string
-}
-
-const initialState: IFormValues = {
-  handle: '',
-  email: '',
-  password: '',
-}
+import theme from '../../theme'
 
 const SignupForm: React.FC = () => {
-  const { handleChange, values, handleValidate, handleBlur, errors, setValues } = useFormValidation<
-    IFormValues,
-    IErrors
-  >(initialState, validateAuth)
-
-  const [createUser, { loading, error }] = useCreateUserMutation({
-    onCompleted: () => {
-      trackCreateAccount()
-      setValues(initialState)
-    },
+  const router = useRouter()
+  const { handleSubmit, register, errors } = useForm({
+    mode: 'onBlur',
   })
 
-  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
-    e.preventDefault()
-    handleValidate(e)
+  const fieldErrorName = Object.keys(errors)[0] || ''
+
+  const [createUser, { loading, error }] = useCreateUserMutation()
+
+  const onSubmit = (data: any) => {
     if (!loading && Object.keys(errors).length === 0) {
       createUser({
         variables: {
-          handle: values.handle,
-          email: values.email,
-          password: values.password,
+          handle: data.handle,
+          email: data.email,
+          password: data.password,
         },
       })
+      trackCreateAccount()
+      router.push({
```

Comment:
Navigating to the next page should stay in the `onCompleted` callback of the `createUser` mutation in case there is an error. I'm also thinking that the `trackCreateAccount` should go there too, since it would look like the user has been created from the event, but an error would prevent the user from being created.

Reply:
Good call, I'll move it in the `LoginForm` as well then 👍🏼

Reply:
Done like dinner! 🍗

Reply:
Done and dusted 🧹 ✨